src/openforms/tests/test_frontend_utils.py

```diff
@@ -0,0 +1,115 @@
+import json
+
+from django.test import TestCase
+
+from furl import furl
+
+from openforms.frontend import get_frontend_redirect_url
+from openforms.submissions.tests.factories import SubmissionFactory
+
+
+class FrontendRedirectTests(TestCase):
+    def test_frontend_redirect_no_hash(self):
+        submission = SubmissionFactory.create(
+            form_url="https://example.com/basepath",
+        )
+
+        action_params = {"action_1": "arg_1", "action_2": "arg_2"}
+
+        redirect_url = get_frontend_redirect_url(
+            submission,
+            "resume",
+            action_params,
+        )
+
+        excpected_redirect_url = furl("https://example.com/basepath")
+        excpected_redirect_url.add(
+            {
+                "_of_action": "resume",
+                "_of_action_params": (json.dumps(action_params)),
```

Comment:
```suggestion
                "_of_action_params": json.dumps(action_params),
```

Reply:
I was thinking black would pick it up when I left it :(